### ISSUE TITLE
Auth action failing 

### ISSUE THREAD

**Issue by harshitaXerox** [google-github-actions/auth#270]
### TL;DR

Hi Team,

the below auth action is failing but the same is running when we hardcode the secret name.
steps.string.outputs.uppercase will have string "METROPLEX"

Could you please let us know what would be the issue.

      - id: 'auth'
        uses: 'google-github-actions/auth@v1'
        with:
#            credentials_json: '${{ secrets.FIREBASE_CAREAR_METROPLEX}}' 
          credentials_json: ${{ secrets[format('FIREBASE_CAREAR_{0}', steps.string.outputs.uppercase )] }}

Thanks
Harshita
[logs_2074.zip](https://github.com/google-github-actions/auth/files/10964913/logs_2074.zip)

### Action YAML

```yaml
- id: 'auth'
        uses: 'google-github-actions/auth@v1'
        with:
#            credentials_json: '${{ secrets.FIREBASE_CAREAR_METROPLEX}}' 
          credentials_json: ${{ secrets[format('FIREBASE_CAREAR_{0}', steps.string.outputs.uppercase )] }}
```

**Comment by sethvargo** [comment]
Hi @harshitaXerox 

This is a general GitHub Actions YAML syntax question. We can only offer support for the `auth` action, but the error is happening during YAML interpolation (which GitHub handles). I found [this Stackoverflow post](https://stackoverflow.com/questions/61255989/dynamically-retrieve-github-actions-secret) which might be helpful. If it's not, I would recommend opening a support ticket with GitHub. Thanks!
